lib/spack/spack/util/executable.py

```diff
@@ -196,28 +225,9 @@ def streamify(arg, mode):
             proc = subprocess.Popen(
                 cmd, stdin=istream, stderr=estream, stdout=ostream, env=env, close_fds=False
             )
-            out, err = proc.communicate()
-
-            result = None
-            if output in (str, str.split) or error in (str, str.split):
-                result = ""
-                if output in (str, str.split):
-                    if sys.platform == "win32":
-                        outstr = str(out.decode("ISO-8859-1"))
-                    else:
-                        outstr = str(out.decode("utf-8"))
-                    result += outstr
-                    if output is str.split:
-                        sys.stdout.write(outstr)
-                if error in (str, str.split):
-                    if sys.platform == "win32":
-                        errstr = str(err.decode("ISO-8859-1"))
-                    else:
-                        errstr = str(err.decode("utf-8"))
-                    result += errstr
-                    if error is str.split:
-                        sys.stderr.write(errstr)
+            out, err = proc.communicate(timeout=timeout)
```

Comment:
Sure, I think subprocess has some builtins for polling stdout/err, should be pretty straightforward.

Reply:
I'm implementing the polling work anyway as it's generally useful, but I don't actually see how this change in particular  would cause us to wait for output anymore than we already would be? `communicate` waits for the process to finish before it returns regardless of whether the timeout argument has been provided, so in the case with no timeout given, we're actually waiting longer for output as `communicate` will take longer to return. 
From the docs:
> Interact with process: Send data to stdin. Read data from stdout and stderr, until end-of-file is reached. Wait for process to terminate.
https://docs.python.org/3/library/subprocess.html?#subprocess.Popen.communicate

Reply:
> For long-running processes, we may want to print output as it comes. 

I misread this code, it was using `communicate` before these changes as well. Marking this as resolved.